tests/pytest/test.py

```diff
@@ -1280,6 +1280,123 @@ def testFilterPrecedence(env):
     r.expect('JSON.GET', 'doc', '$[?(@.f==true || @.one==1 && @.t==false)]').equal('[]')
 
 
+def testOutOfRangeValues(env):
+    # Test numeric min/max values
+    r = env
+
+    doc_int_ok = {
+        "beyond_max_i64": 9223372036854775808, # as u64
+        "min_i64": -9223372036854775808,
+        "below_min_i64": -9223372036854775809, # as f64
+        "max_u64": 18446744073709551615,
+    }
+
+    doc_int_ok_overflow_mult = {
+        "beyond_max_u64": 18446744073709551616, # as f64
+    }
+
+    doc_int_ok_overflow_both = {
+        "max_i64": 9223372036854775807,
+    }
+
+    doc_float_ok = {
+        "max_f64" : 1.7976931348623157e+308,
+        "max2_f64": 1.7976931348623158e+308,
+        "min_f64" : -1.7976931348623157e+308,
+        "min2_f64": -1.7976931348623158e+308,
+    }
+
+    doc_float_ok_overflow_mult = {
+        "max_u64": 18446744073709551615,
+    }
+
+    doc_float_ok_2_overflow_mult = {
+        "max_i64": 9223372036854775807.0,
+        "beyond_max_i64": 9223372036854775808.0, # as u64
+        "min_i64": -9223372036854775808.0,
+        "below_min_i64": -9223372036854775809.0, # as f64
+        "max_u64": 18446744073709551615.0,
+    }
+
+    doc_bad_values = {
+        "beyond_max_f64": 1.7976931348623159e+308,
+        "below_min_f64": -1.7976931348623159e+308
+    }
+    
+    r.expect('JSON.SET', 'doc_int_ok', '$', json.dumps(doc_int_ok)).ok()
+    r.expect('JSON.SET', 'doc_int_ok_overflow_mult', '$', json.dumps(doc_int_ok_overflow_mult)).ok()
+    r.expect('JSON.SET', 'doc_int_ok_overflow_both', '$', json.dumps(doc_int_ok_overflow_both)).ok()
+
+    r.expect('JSON.SET', 'doc_float_ok', '$', json.dumps(doc_float_ok)).ok()
+    r.expect('JSON.SET', 'doc_float_ok_overflow_mult', '$', json.dumps(doc_float_ok_overflow_mult)).ok()
+
+    r.expect('JSON.SET', 'doc_float_ok_2_overflow_mult', '$', json.dumps(doc_float_ok_2_overflow_mult)).ok()
+    
+    def check_object_values(obj, name, epsilon, check_num_incr=True, check_num_mult=True):
+        # Test values from JSON.GET are equal to JSON.SET
+        # Check no crash (using get_long internally)
+        arr = []
+        for k, v in iter(obj.items()):
+            r.assertTrue(True, message='{} GET {}={}'.format(name, k, v))
+            res = r.execute_command('JSON.GET', name, '$.{}'.format(k))
+            r.assertAlmostEqual(json.loads(res)[0], v, epsilon, message=res)
+            try:
+                # Check no crash on overflow
+                res = r.execute_command('JSON.NUMINCRBY', name, '$.{}'.format(k), '1')
+                if check_num_incr:
+                    r.assertAlmostEqual(json.loads(res)[0], v+1, epsilon, message=res)
+                r.execute_command('JSON.NUMINCRBY', name, '$.{}'.format(k), '9223372036854775808')
```

Comment:
I am not follow, if one of the checks raise error we are not checking the rest? What am I missing?

Reply:
Right, seemed enough to check first overflow/unsupported value, but we can test all.